lib/shipit/same_site_cookie_middleware.rb

```diff
@@ -0,0 +1,31 @@
+module Shipit
+  class SameSiteCookieMiddleware
+    COOKIE_SEPARATOR = "\n".freeze
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      status, headers, body = @app.call(env)
+
+      if headers && headers['Set-Cookie'] &&
+         env['SHIPIT_ENABLE_SAMESITE_NONE'].present? &&
```

Comment:
Two questions:
  - Is there any reason to gate this behind an env variable?
  - If yes, then wouldn't it be simpler to just not insert the middleware?

Reply:
Fair point! To me it makes sense to make the feature opt-in, since some users may want to prevent their cookies from being used externally. I agree that the check would be better at the middleware level, though.

Happy to defer to your judgement on whether the check is necessary at all - let me know.

Reply:
Right, since we never figured a way to make the browser extension available, I suppose the vast majority of users don't need this middleware.

Maybe we should keep it a host app responsability?

Reply:
Agreed - I'll keep this an opt-in feature via the environment variable, but I'll have it enable the middleware instead of check on each request.